ntbs-integration-tests/ServiceDirectory/ServiceDirectorySearchTest.cs

```diff
@@ -234,16 +235,51 @@ private static int GetNumberOfDisplayedResults(IHtmlDocument pageContent)
-            var expectedServiceResults = (await referenceDataRepository.GetActiveTBServicesBySearchKeywords(searchKeywords)).Count;
-            var expectedHospitalResults = (await referenceDataRepository.GetActiveHospitalsBySearchKeywords(searchKeywords)).Count;
-            var expectedUserResults = (await userSearchService.OrderQueryableAsync(searchKeywords)).Count;
+            var expectedRegionResults = await context.PHEC
+                .CountAsync(x => x.Name.ToLower().Contains(searchKeyword));
+            var expectedServiceResults = await context.TbService
+                .Include(t => t.PHEC)
+                .CountAsync(t => t.Name.ToLower().Contains(searchKeyword) && !t.IsLegacy);
+            var expectedHospitalResults = await context.Hospital
+                .Include(h => h.TBService.PHEC)
+                .CountAsync(h => h.Name.ToLower().Contains(searchKeyword) && !h.IsLegacy);
+            var expectedUserResults = await GetExpectedUserResults(context, searchKeywords);
 
             return expectedRegionResults + expectedServiceResults + expectedHospitalResults + expectedUserResults;
         }
+
+        private static async Task<int> GetExpectedUserResults(NtbsContext context, List<string> searchKeywords)
+        {
+            // This method is a replication of the UserSearchService which bypasses all calls to the
+            // referenceDataRepository service
+            
+            var allPhecs = await context.PHEC.OrderBy(x => x.Name).ToListAsync();
+            var filteredPhecs = allPhecs
+                .Where(phec => searchKeywords.All(s => phec.Name.ToLower().Contains(s)));
+
+            var caseManagersAndRegionalUsers = (await context.User
+                    .Include(u => u.CaseManagerTbServices)
+                    .ThenInclude(c => c.TbService)
+                    .ThenInclude(tb => tb.PHEC)
+                    .OrderBy(u => u.DisplayName)
+                    .ToListAsync())
+                .Where(u => u.IsActive && (u.CaseManagerTbServices.Any()
+                                           || (u.AdGroups != null &&
+                                               allPhecs.Any(phec => u.AdGroups.Split(",").Contains(phec.AdGroup)))));
+            
+            var filteredCaseManagersAndRegionalUsers = caseManagersAndRegionalUsers.Where(c =>
+                    searchKeywords.All(s =>
+                        c.FamilyName != null && c.FamilyName.ToLower().Contains(s)
+                        || c.GivenName != null && c.GivenName.ToLower().Contains(s))
+                    || searchKeywords.All(s => c.DisplayName != null && c.DisplayName.ToLower().Contains(s))
+                    || c.CaseManagerTbServices.Any(x =>
+                        searchKeywords.All(s => x.TbService.Name.ToLower().Contains(s)))
+                    || filteredPhecs.Any(phec => c.AdGroups != null && c.AdGroups.Split(",").Contains(phec.AdGroup)))
```

Comment:
The fact that you've included this here has actually alerted me to the fact that I think you've missed a requirement - user search should no longer be returning users where the TB service or region match - we only want users where the name matches. This should be in the ACs on the ticket (if it isn't then please let me know and I'll add it).

Reply:
I hadn't interpreted it this way - I'll make the change though :)